### ISSUE TITLE
Pretty printing produces invalid json

### ISSUE THREAD

**Issue by pwinckles** [avaje/avaje-jsonb#132]
Example:

```java
import io.avaje.jsonb.Json;
import io.avaje.jsonb.Jsonb;

import java.util.List;

@Json
public record PrettyExample(String name, List<String> values, int count) {

    public static void main(String[] args) {
        var jsonb = Jsonb.builder().build();
        var type = jsonb.type(PrettyExample.class);
        var example = new PrettyExample("test", List.of("one", "two"), 2);
        System.out.println("Normal output is correct: " + type.toJson(example));
        System.out.println("Pretty print output is invalid json and oddly formatted: " + type.toJsonPretty(example));
        System.out.println("Somehow invalid json can be parsed: " + type.fromJson(type.toJsonPretty(example)));
    }

}
```

Produces:

```
Normal output is correct: {"name":"test","values":["one","two"],"count":2}
Pretty print output is invalid json and oddly formatted: {
  "name": "test",
  "values":   [
"one","two"    ]

  "count": 2
}
Somehow invalid json can be parsed: PrettyExample[name=test, values=[one, two], count=2]
```

Note that in addition to producing invalid json, the formatting is odd, and it can somehow be round tripped.

**Comment by SentryMan** [comment]
you're on a roll today, I'll take a look

**Comment by SentryMan** [comment]
@rbygrave I'm just a little out of my depth here since it happens deep in the parser.

**Comment by rbygrave** [comment]
FYI: The fix is in 1.6-RC6 if you want to try that out.

**Comment by SentryMan** [comment]
Hmm, that may be harder to fix without slowing everything down. As I understand it, after the parser reads an ending token `]`/`}` or closing `"` after a value, it immediately looks for the next instance of double quotes, rather than a comma. As a side-effect, jsonb can parse and deserialize this without issues.
```
      	{
      	    "name": "test"
      	    "values": [
      	        "one"
      	        "two"
      	    ],
      	    "count": 2,
      	}
```

but super invalid Json like ```      	
{
      	    "name": "test",,,
      	    "values": [
      	    //comments
      	        "one"
      	        "two"
      	    ],
      	    "count": 2
      	}
```
will naturally fail to parse.

Since it's only an issue(?) with adding/forgetting single trailing commas I personally am willing to write this off as a feature, what say you?

**Comment by pwinckles** [comment]
On one hand, I'm struggling to think of an ambiguity that necessitates commas between fields at all, and on the other hand it seems a little unexpected to me to be able to parse and invalid document. For example, let's say I was using jsonb to serialize pretty-printed documents. Before this fix, I was generating invalid documents, but had no way of knowing this without physically examining them because jsonb would roundtrip them no problem. This could then become an issue if the documents were to be consumed by another system that accepts only valid json. That said, it is only real concern so far as its possible for jsonb to generate invalid json in the first place.

**Comment by rob-bygrave** [comment]
We can treat this as a bug, I'll log it as a bug.

>  I'm struggling to think of an ambiguity that necessitates commas between fields at all

avaje-jsonb has support for `x-json-stream` which allows for "new line delimited arrays" (which means the commas are expected to be missing).

The effective bug here is that it is using the same logic in `hasNextElement()` to both support "Normal Arrays" and to support "x-json-stream / new line delimited arrays".  Not that technically with `x-json-stream` the `[` and `]` are also allowed to not be included as well as the commas hence there is `beginStream()` and `endStream()` but effectively there is a missing `hasNextStreamElement()`.

**The effective fix** which I have locally adds a `hasNextStreamElement()` which will be used for `x-json-stream` and thus the `hasNextElement()` is tightened up to **_expect_** the comma or `]` and will throw a parsing exception when that isn't the case.

TLDR: Proper separation between processing "normal arrays" and "x-json-stream / new line delimited arrays"

**Comment by rob-bygrave** [comment]
Logged as: https://github.com/avaje/avaje-jsonb/issues/138
